File: frontend/quality-platform/src/types/trace.ts
// Mirrors the backend AIQuality.Core.Entities.Trace shape.
export type TraceStatus = 0 | 1 | 2 | 3; // Pending | Running | Completed | Failed

export const TRACE_STATUS_LABEL: Record<TraceStatus, string> = {
  0: "Pending",
  1: "Running",
  2: "Completed",
  3: "Failed",
};

export interface Trace {
  id: string;
  name: string;
  model: string;
  status: TraceStatus;
  startedAt: string;
  endedAt: string | null;
  durationMs: number;
  qualityScore: number | null;
}

export interface StartTraceRequest {
  name: string;
  model: string;
}

File: backend/AIQuality.Core/Entities/Trace.cs
using AIQuality.Core.Enums;

namespace AIQuality.Core.Entities;

// A distributed trace for a single AI request (one model invocation end-to-end).
public class Trace
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public TraceStatus Status { get; set; } = TraceStatus.Pending;
    public DateTimeOffset StartedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? EndedAt { get; set; }
    public long DurationMs { get; set; }
    // Optional quality score (0..1) attached by the evaluator once judged.
    public double? QualityScore { get; set; }
}
